### ISSUE TITLE
how to merge the weights of uvtr_lidar and uvtr_cam

### ISSUE THREAD

**Issue by hbzhou530** [Nightmare-n/UniPAD#19]
Thank you for your meaningful work. How to get merged_uvtr_fusion_vs0.075_finetune.pth

**Comment by Nightmare-n** [comment]
Please use the following pseudo-code to merge the weights:
```
image_branch = torch.load('uvtr_cam_xxx.pth', map_location='cpu')
lidar_branch = torch.load('uvtr_lidar_xxx.pth', map_location='cpu')
fusion = {}
fusion['state_dict'] = {**image_branch['state_dict'], **lidar_branch['state_dict']}
torch.save(fusion, 'uvtr_fusion_xxx.pth')
```

**Comment by Song-Jingyu** [comment]
Hi,

Thanks for the detailed introduction on how to merge the weights. I followed the process and get the merged weight to launch finetuning final_uvtr_fusion_vs0.075_finetune. For merging I was using the official weights of camera and LiDAR stream from the Google drive.

After the pretraining I only obtained NDS 71.7 and mAP 67.3, which is still a bit far from the reported results. I would really appreciate if there is any insight on why I might get this! I am using 4xA100 GPU and the same config (similar batch size and lr). Thank you!
